gdxsv/lbs_discord.go

```diff
@@ -0,0 +1,525 @@
+		chunkSize := 25
+
+		for i := 0; i < len(fields); i += chunkSize {
+			end := i + chunkSize
+
+			if end > len(fields) {
+				end = len(fields)
+			}
+
+			divided = append(divided, fields[i:end])
+		}
+		return divided
+	}
+
+	reduceJSONStringSize := func(s string) string {
+		//Reduce size by removing userid
+		re := regexp.MustCompile("`.*?`\\s")
+		s = re.ReplaceAllString(s, "")
+
+		replacer := strings.NewReplacer("<:gundam:772467554160738355>", "🌎", "<:zaku:772467605008023563>", "🪐")
+		s = replacer.Replace(s)
+
+		return s
+	}
+
+	//
+	// Implementation for the core function
+	//
+	publish = func() {
+		//Stop any retry request since new request appeared
+		if discordLiveStatusRetryTimer != nil {
+			discordLiveStatusRetryTimer.Stop()
+			discordLiveStatusRetryTimer = nil
+		}
+
+		//
+		// Create battle user list, for the third embed
+		//
+		battle := make(map[string]*battleUsers)
+		existingBattleUserIDs := make(map[string]bool)
+
+		battleUserCount := 0
+		accumulatedEmbedStringLength := 0
+
+		for _, u := range sharedData.GetMcsUsers() {
+			_, exists := battle[u.BattleCode]
+			if exists == false {
+				battle[u.BattleCode] = new(battleUsers)
+
+				locName, ok := gcpLocationName[u.McsRegion]
+				if !ok {
+					locName = "Default Server"
+				}
+				if u.McsRegion == "best" {
+					locName = "Best Server"
+				}
+				battle[u.BattleCode].RegionName = locName
+				accumulatedEmbedStringLength += len(locName)
+			}
+
+			var user string
+			switch u.Side {
+			case TeamRenpo:
+				user = fmt.Sprintf("<:gundam:772467554160738355> `%s` %s\n", u.UserID, u.Name)
+				battle[u.BattleCode].RenpoUsersString += user
+			case TeamZeon:
+				user = fmt.Sprintf("<:zaku:772467605008023563> `%s` %s\n", u.UserID, u.Name)
+				battle[u.BattleCode].ZeonUsersString += user
+			}
+			accumulatedEmbedStringLength += len(user)
+
+			battleUserCount++
+			existingBattleUserIDs[u.UserID] = true
+		}
+
+		//
+		// Create lobby user list, for the second embed
+		//
+		var plazaUsers string
+
+		lobby := make(map[uint16]*lobbyUsers)
+
+		plazaUserCount := 0
+		lobbyUserCount := 0
+
+		lbs.Locked(func(lbs *Lbs) {
+			for _, u := range lbs.userPeers {
+
+				//Already in battle, hidden from lobby
+				if existingBattleUserIDs[u.UserID] {
+					continue
+				}
+
+				if u.Lobby == nil {
+					plazaUsers += fmt.Sprintf("`%s` %s\n", u.UserID, u.Name)
+					accumulatedEmbedStringLength += len(plazaUsers)
+					plazaUserCount++
+				} else {
+					_, exists := lobby[u.Lobby.ID]
+					if exists == false {
+						lobby[u.Lobby.ID] = new(lobbyUsers)
+						lobby[u.Lobby.ID].Name = u.Lobby.Name
+
+						locName, ok := gcpLocationName[u.Lobby.McsRegion]
+						if !ok {
+							locName = "Default Server"
+						}
+						if u.Lobby.McsRegion == "best" {
+							locName = "Best Server"
+						}
+						lobby[u.Lobby.ID].RegionName = locName
+
+						var comment string
+						if strings.Contains(u.Lobby.Comment, "TeamShuffle") {
+							comment += "🔀"
```

Comment:
Do not use comment strings in logic.
This is likely to change easily, but we don't know that it's broken at compile time.

Reply:
Updated to use lobby settings for `TeamShuffle`, `No375MS`, `AutoReBattle`, but for "For JP vs HK" / "Private Room", it is in the "if it is going to break, let it break" state.